tests_e2e/tests/ext_policy/ext_policy_with_dependencies.py

```diff
@@ -0,0 +1,282 @@
+                    try:
+                        if deployment_failure_match is None:
+                            raise Exception("Unable to match a ResourceDeploymentFailure")
+                        error_json = json.loads(deployment_failure_match.group("error"))
+                        error_message = error_json['message']
+                    except Exception as parse_exc:
+                        fail("Extension template deployment failed as expected, but there was an error in parsing the failure. Parsing failure: {0}\nDeployment Failure: {1}".format(parse_exc, e))
+
+                    for phrase in expected_errors:
+                        if phrase not in error_message:
+                            fail("Extension template deployment failed as expected, but with an unexpected error. Error expected to contain message '{0}'. Actual error: {1}".format(phrase, e))
+                log.info("Extensions failed as expected")
+                log.info("")
+
+            # After each test, clean up failed extensions to leave VMSS in a good state for the next test.
+            # If there are leftover failed extensions, CRP will attempt to uninstall them in the next test, but uninstall
+            # will be disallowed by policy. Since CRP waits for a 90 minute timeout for uninstall, the operation will
+            # timeout and fail without an appropriate error message (known issue), and the whole test case will fail.
+            # To clean up, we first update the policy to allow all, then remove the extensions.
+            log.info("Starting cleanup for test case...")
+            for ssh_client in ssh_clients.values():
+                allow_all_policy = \
+                    {
+                        "policyVersion": "0.1.0",
+                        "extensionPolicies": {
+                            "allowListedExtensionsOnly": False
+                        }
+                    }
+                self._create_policy_file(ssh_client, allow_all_policy)
+
+            for ext_to_delete in deletion_order:
+                ext_name_to_delete = ext_to_delete.type
+                try:
+                    self._context.vmss.delete_extension(ext_name_to_delete)
+                except Exception as crp_err:
+                    # Known issue - CRP returns stale status in cases of dependency failures. Even if the deletion succeeds,
+                    # CRP may return a failure here. We swallow the error, and instead, check that the logs for uninstall
+                    # are present in the agent log (after the start time of this test case).
+                    log.info("CRP returned an error for deletion operation, may be a false error. Checking agent log to determine if operation succeeded. Exception: {0}".format(crp_err))
+                    try:
+                        for ssh_client in ssh_clients.values():
+                            msg = ("Remove the extension slice: {0}".format(str(ext_to_delete)))
```

Comment:
This message is related to cgroup. Right now it's logged even when cgroup isn't enabled (which might and probably should change in the future).

Instead, we should check that the handler was successfully uninstalled. i.e. the last ext status reported by the agent shouldn't include the handler:

> 2024-11-26T23:54:04.306568Z INFO ExtHandler ExtHandler Extension status: [('Microsoft.Azure.Monitor.AzureMonitorLinuxAgent', 'Ready')]

You might also consider doing this by checking the instance view

Reply:
Updated this - we now check for the last status reported by the agent and confirm that there is no handler status reported. (Instance view doesn't work because CRP reports a stale status)